docs/source/user_guide/key_concepts.md

```diff
@@ -0,0 +1,175 @@
+
+In this figure you can see on the left a schematic of a line-scanning microscope acquiring an image of a grid. The scanning pattern plus the sample rotation lead fan-like aftefacts. On the right you see a grid that has been imaged while still (top), while rotating (middle) and the derotated image (bottom). The grid is now perfectly aligned.
+```
+
+
+If the angle of rotation is recorded, `derotation` can reconstruct each frame by assigning a rotation angle to each acquired line and rotating it back to its original position. This is what is called derotation-by-line. This process incrementally reconstructs each frame and can optionally include shear deformation correction.
+
+
+```{figure} ../_static/derotation_by_line.gif
+
+In this animation you can see the reconstruction of a frame with derotation. The original frame with deformation is on the left. With subsequent iterations, derotation picks a line, rotates it according to the calculated angle, and adds it to the new derotated frame. The final result is on the right.
+```
+
+
+## How to use derotation
+
+There are two main ways to use `derotation`:
+
+### 1. **Low-level core function**
+Use {func}`derotation.derotate_by_line.derotate_an_image_array_line_by_line` to derotate an image stack, given:
+- The original multi-photon movie (expects only one imaging plane)
+- Rotation angle per line
+
+This is ideal for testing and debugging with synthetic or preprocessed data.
+
+Please refer to the [example on how to use the core function](../examples/core_fucntion.rst) for a demonstration of how to use {func}`derotation.derotate_by_line.derotate_an_image_array_line_by_line`.
+
+### 2. **Full and Incremental Pipeline Classes**
+
+Derotation provides two pre-built pipelines for end-to-end processing:
+
+- **{class}`derotation.analysis.full_derotation_pipeline.FullPipeline`**
+  Assumes randomized, alternating clockwise and counter-clockwise rotations. It performs:
+  - Analog signal parsing
+  - Angle interpolation
+  - Bayesian optimization for center estimation
+  - Line-by-line derotation using {func}`derotation.derotate_by_line.derotate_an_image_array_line_by_line`
+
+- **{class}`derotation.analysis.incremental_derotation_pipeline.IncrementalPipeline`**
+  Assumes a continuous rotation performed in small increments. Inherits from `FullPipeline` but skips Bayesian optimization.
+  Useful when the center of rotation is known or estimated differently.
+
+Both pipelines accept a **configuration dictionary** (see the [configuration guide](./configuration.md)) and produce:
+- A derotated TIFF stack
+- A CSV file with rotation angles and metadata
+- Debugging plots
+- A text file containing the estimated optimal center of rotation
+- Log files
+
+You can create custom pipelines by subclassing `FullPipeline` and overriding the relevant methods.
+
+See the [usage example](../examples/pipeline_with_real_data.zip) for how to instantiate `FullPipeline`, run it, and inspect its attributes and outputs.
```

Comment:
A lot of people won't be happy to download a random zip archive. Also, I would host this data on GIN rather than in the git repo.

Reply:
That's actually a typo, there's no zip file!